### PR TITLE
GUI: Fix decimator zoom by single axis (#712)

### LINKED ISSUE #712: BUG- Horizontal and vertical zoom do not reapply decimation
**Description**
The chart does not reapply decimation or have issue zooming in the chart. It is not possible to zoom in one way, and the other zoom will not repopulate the chart.

**How to reproduce**
```python
from taipy.gui import Gui
from math import cos, exp
from taipy.gui.data.decimator import MinMaxDecimator

def compute_data(nb_points)->list:
    return {"x":range(nb_points), "y":[cos(i/6) * exp(-i*decay/nb_points) for i in range(nb_points)]}


decay = 0
nb_points = 10_000
decimator_instance = MinMaxDecimator(n_out=1_000, zoom=True)

data = compute_data(nb_points)

page = """
# Taipy *Demo*

The nb_points is: <|{nb_points}|>.

<|{nb_points}|selector|lov={[10_000, 20_000, 50_000, 100_000, 200_000, 500_000, 1_000_000]}|dropdown|on_change=change|>

<|{data}|chart|x=x|y=y|decimator=decimator_instance|>
"""

def change(state):
    state.data = compute_data(state.nb_points)


Gui(page).run(port=5020, use_reloader=True)
```

- Zoom vertically or horizontally in the chart.

image

It is not possible to zoom in one way and the other will not repopulate the chart.

This is the expected result:

image

**Expected behavior**
The chart should be repopulated when the user zoom vertically or horizontally on the chart.

**Runtime environment**
Taipy GUI: develop for 3.1

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #712

### CODE REVIEW

File: taipy/gui/data/utils.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #        http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
-# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR df_filter_conditions OF ANY KIND, either express or implied. See the License for the
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/utils.py]: Not sure about this change 🤔

**Review by FredLL-Avaiga — DISMISSED** [review]
DrawifyGoalGIF

**Review by FabienLelaquais — DISMISSED** [review]
Who would have thought?

File: taipy/gui/data/utils.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/data/utils.py]: I have fixed it though :D

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx [hunk 2]
```diff
@@ -512,38 +512,35 @@ const Chart = (props: ChartProp) => {
...
-                            dtKey,
-                            getDecimatorsPayload(
-                                config.decimators,
-                                plotRef.current,
-                                config.modes,
-                                config.columns,
-                                config.traces,
-                                eventData
-                            )
+            onRangeChange && dispatch(createSendActionNameAction(id, module, { action: onRangeChange, ...eventData }));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Chart.tsx:515]: good catch

**Review by FredLL-Avaiga — APPROVED** [review]

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx (hunk 2 above)
Comment (dinhlongviolin1) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx:515]: tx @FredLL-Avaiga

**Pull request merged**